index.bs

```diff
@@ -1527,6 +1545,40 @@ example of the latter, when the user is accessing the [=[RP]=] from a given clie
 use a [=roaming authenticator=] which was originally registered with the [=[RP]=] using a different client.
 
 
+### Attestation Presentation Preference enumeration (enum <dfn enum>AttestationPresentationPreference</dfn>) ### {#attestation}
+
+[=[RPS]=] may use the {{AttestationPresentationPreference}} enum to specify their preference regarding attestation types used
+during credential generation.
+
+<pre class="idl">
+    enum AttestationPresentationPreference {
+        "none",
+        "indirect",
+        "direct"
+    };
+</pre>
+
+<ul>
+    <li><dfn>none</dfn> - indicates that the [=[RP]=] is not interested in attestation, and prefers fast and reliable credential
+      creation over obtaining a verifiable attestation. The client MAY replace the attestation statement generated
+      by the authenticator with a meaningless client-generated attestation statement in this case, for example in order to prevent
+      having to obtain user consent for relaying identifying information to the RP, or to save a roundtrip to a Privacy CA. 
+
+      This is the default value.
+    <li><dfn>indirect</dfn> - indicates that the [=[RP]=] prefers an attestation type that leads to verifiable attestation
+      statements, but leaves it up to the client how to obtain such attestation statements. The client MAY replace the
+      authenticator-generated attestation statements with attestation statements generated by a Privacy CA, in order to protect
+      the privacy of the user, or to assist [=RPS=] with attestation verification in a heterogeneous ecosystem.
+
+      Note that there is no guarantee that the [=RP=] will obtain a verifiable attestation statement in this case. (For
+      example, in the case that the authenticator uses self-attestation.)
+    <li><dfn>direct</dfn> - indicates that the [=[RP]=] wants the attestation statement as generated by the authenticator. If
+      the client returns an attestation statement to the RP in this case, it MUST be the statement generated by the authenticator.
+      If, for whatever reason, the client cannot pass on the authenticator-generated attestation statement, it MUST terminate
```

Comment:
we would prefer in the direct case to simply be informed as to any issues the client may know of wrt the authnr, but still receive the attestation. Also, a processing requirment such as this needs to be in the relevant algorithm rather than here.

Reply:
Responding in two places to make sure this is not lost:

Agree with @rlin1 concern. I'd like to suggest the following re-wording:

`If authenticator violates privacy requirements of the attestation scheme it is using, the client MUST terminate the credential generation operation with a AttestationNotPrivateError' 

In my opinion, this wording scopes the problem to privacy and at the same time extends it beyond batch attestation issues. For example, improper implementation of the TCG attestation scheme could also be covered.

Reply:
Feel free to make a commit to this PR to reflect your changes.

Reply:
Done. https://github.com/w3c/webauthn/pull/636/commits/825cce701d532364a78c7ace2918d1a6a64a0641

@rlin1 / @equalsJeffH Is this description ok?